lib/searchers/mongo.js

```diff
@@ -65,11 +67,23 @@ var methods = {
       cursorOptions.skip = options.skip;
     }
 
+    cursorOptions.limit = options.limit;
+
     cursor = index.collection.find(selector, cursorOptions);
 
+    // Get the total count by aggregating
+    pipeline = [
+      {$match: selector},
+      {$group: {
+        _id: "id",
+        total: {$sum: 1}
+      }}
+    ];
+    aggregates = index.collection.aggregate(pipeline);
+
     results = {
-      'results' : _.first(cursor.fetch(), options.limit),
-      'total' : cursor.count()
+      'results' : cursor.fetch(),
+      'total' : aggregates.length >= 1 ? aggregates[0].total : 0
```

Comment:
so, the tests are failing because on the client side it doesn't have an aggregate method. This can be solved by having an `if (Meteor.isServer)` clause that checks on which environment the code is run on. If it's on the client, it should use the old behaviour.

Can you fix that? I'd love to improve the performance with this.